VelocidiSDK/VelocidiSDK/VSDKUserId.m

```diff
@@ -2,10 +2,31 @@
 
 @implementation VSDKUserId
 
-- (instancetype)initUserId: (NSString *)userId
-                          : (NSString *)type  {
-    if(self = [super init]){
-        self.type = type;
+- (instancetype)init {
+    @throw nil;
+}
+
+- (instancetype)initWithId: (NSString *)userId
+                      type: (NSString *)userIdType {
+    
+    if ([userId length] <= 0) {
+        NSException* ex = [NSException
+                exceptionWithName:@"InvalidArgument"
+                reason:@"id must not be empty!"
+                userInfo:nil];
+        @throw ex;
+    }
+    
+    if ([userIdType length] <= 0) {
+        NSException* ex = [NSException
+                exceptionWithName:@"InvalidArgument"
+                reason:@"type must not be empty!"
+                userInfo:nil];
+        @throw ex;
+    }
```

Comment:
I don't think we should be throwing these exceptions here. Perhaps we can do these verifications when we are executing the request and return these errors in the `onFailureBlock`.

Reply:
In 96af9d1 I try to refactor things to avoid throwing exceptions.